ystia/yorc
# Feature/brbdcf 555 ssh key based authentication slurm

# Pull Request description

## Description of the change

Added SSH Key-based authentication for slurm infrastrutures

### What I did

Parsing slurm configuration options and expecting password or private_key to be defined, else an error is returned.
Added an authentication method based on the private key when it is defined.
Documented this new option.
Added unit tests.

### How I did it

### How to verify it

On a host where the orchestrator is running, login as user yorc and generate private/public keys using the following command, keeping all default values (no passphrase):
$ ssh-keygen

Change generated files permissions:
$ chmod 400 $HOME/.ssh/id_rsa*

Copy the content of public key $HOME/.ssh/id_rsa.pub to your non-root user home directory on the Slurm client node in file <home of the non-root user>/.ssh/authorized_keys

Use the following configuration file to launch the orchestrator:
/etc/yorc/config.yorc.json:
{
  "log": 1,
  "infrastructures": {
    "slurm": {
      "user_name": "your_non_root_user",
      "private_key": "york_user_home_directory/.ssh/id_rsa",
      "url": "slurm _client_node_IP_address",
      "port": 22
    }
  }
}

Launch alien4cloud, load Yorc plugin, create the corresponding orchestrator and its slurm location.
Deploy an application.
Check as your non-run user on Slurm client node, that the command 'squeue' shows a job running as your non-root user and corresponding to your deployment.

### Description for the changelog

## Applicable Issues
